packages/block-library/src/image/edit.js

```diff
@@ -152,6 +153,16 @@ class ImageEdit extends Component {
 				captionFocused: false,
 			} );
 		}
+
+		if ( this.state.autoFocus && this.container ) {
+			this.setState( {
+				autoFocus: false,
+			} );
+			const caption = this.container.querySelector( 'figcaption' );
```

Comment:
Ideally we'd just assign the intended focus target as the `ref`. This is blocked by #9740. It seems like it'd be reasonable enough for `RichText` to use `forwardRef` to make the underlying element available for reference.